bx_py_utils/test_utils/snapshot.py

```diff
@@ -75,6 +75,23 @@ def _get_caller_names(root_dir: Union[pathlib.Path, str] = None, snapshot_name:
     return root_dir, snapshot_name
 
 
+def _get_snapshot_file(
+    root_dir: Union[pathlib.Path, str] = None,
+    snapshot_name: str = None,
+    extension: str = None,
+):
+    # Auto set dir/name via stask information, if not set:
+    root_dir, snapshot_name = _get_caller_names(root_dir, snapshot_name)
+
+    assert re.match(r'^[-_.a-zA-Z0-9]*$', extension), f'Invalid extension {extension!r}'
+
+    if not snapshot_name.endswith('.snapshot'):
```

Comment:
This is backwards-incompatible, isn't it?

We still have callers who use names without `.snapshot` at the end, right? These would suddenly be switched.

Reply:
Yes but current implementation is not useable for auto generated names, because `.snapshot` will be added two times :(